vapour_api/api.py

```diff
@@ -0,0 +1,168 @@
+    )
+    route_stop_positions = []
+    for row in result:
+        s = {
+            "route_position_id": row[0],
+            "geojson": ast.literal_eval(row[1]),
+            "lat": row[2],
+            "lon": row[3],
+        }
+        route_stop_positions.append(s)
+    return route_stop_positions
+
+def get_route_stops_with_connections(route_id):
+    result = db.engine.execute(
+        """
+            SELECT distinct i_positions.member_osm_id, i_positions.member_type, 
+                i_positions.member_index, 
+                i_stops.name,
+                st_asGeoJSON(ST_Transform(i_positions.geom, 4326)), 
+                st_X(ST_Transform(i_positions.geom, 4326)) as lon,
+                st_Y(ST_Transform(i_positions.geom, 4326)) as lat,
+                array_agg(distinct 
+                    array_append(
+	                    array_append(
+	                    	array_append(Array[]::text[], other_routes.osm_id || ''),
+		                    	other_routes.ref || ''
+	                	),
+                    	other_routes.colour || ''
+                	)
+                ) 
+            FROM i_positions 
+            	INNER JOIN i_stops on i_positions.member_osm_id = i_stops.osm_id
+                LEFT JOIN i_positions AS other_positions ON
+                    other_positions.member_type = i_positions.member_type AND
+                    other_positions.member_osm_id = i_positions.member_osm_id AND
+                    other_positions.rel_osm_id != {route_id}
+                LEFT JOIN i_routes AS other_routes ON
+                    other_routes.osm_id = other_positions.rel_osm_id
+            WHERE i_positions.rel_osm_id = {route_id}
+            GROUP BY i_positions.member_osm_id, i_positions.member_type, i_positions.member_index, 
+            	i_stops.name, i_positions.geom
+            ORDER BY member_index
+        """.format(route_id=route_id)
+    )
+    route_stops = []
+    for row in result:
+        s = {
+            "member_osm_id": row[0],
+            "member_type": row[1],
+            "name": row[3],
+            "geojson": ast.literal_eval(row[4]),
+            "lon": row[5],
+            "lat": row[6],
+            "shields": [{"ref": f[1], "colour": f[2]} for f in sorted(row[7], key=lambda x: x[1]) if f[1]],
+        }
+        route_stops.append(s)
+    return route_stops
+
+class Index(Resource):
+    def get(self):
+        return {
+            "links": [
+                {
+                    "href": "api.py/route/<route_id>"
+                }
+            ]
+        }
+            
+class Route(Resource):
+    def get(self, route_id):
+        if route_id:
+            r = OrderedDict([])
+            route_properties = get_route_properties(route_id)
+            r["route_info"] = route_properties
+            route_stop_positions = get_route_stop_positions(route_id)
+            r["stop_positions"] = [{'lat': sp["lat"], 'lon': sp["lon"]} for sp in route_stop_positions]
```

Comment:
It would be better to get a geojson with the array of route stop positions here.

Reply:
Do you want a `GeometryCollection` of Points ? 
```json
{ 
    "type": "GeometryCollection",
    "geometries": [    
        { 
            "type": "Point",
            "coordinates": [100.0, 0.0]
        },
        { 
            "type": "Point",
            "coordinates": [100.0, 0.0]
        }
    ]
}
```

Reply:
yes, it will work with a GeometryCollection of points.